app/services/audit/checksum_validator.rb

```diff
@@ -8,6 +8,8 @@ class ChecksumValidator
     delegate :moab_storage_root, :preserved_object, to: :moab_record
     delegate :storage_location, to: :moab_storage_root
     delegate :druid, to: :preserved_object
+    delegate :moab_on_storage_absent?, :latest_moab_storage_object_version, :object_dir, :moab_on_storage_validator,
+             to: :directory_checksum_validator
```

Comment:
Move storage/directory concerns into newly extracted class to validate moab storage objects whether they are cataloged or not.